Sources/TuistGraph/Graph/GraphOutputJSON.swift

```diff
@@ -0,0 +1,142 @@
+    
+    public static func from(_ graph: Graph, withConfig config: Config) -> GraphOutputJSON {
+        var projects = [String: ProjectOutputJSON]()
+        for (_, project) in graph.projects {
+            // Convert packages.
+            var packages = [PackageOutputJSON]()
+            for package in project.packages {
+                switch package {
+                case .remote(let url, _):
+                    packages.append(PackageOutputJSON(kind: PackageOutputJSON.PackageKind.remote, path: url))
+                case .local(let path):
+                    packages.append(PackageOutputJSON(kind: PackageOutputJSON.PackageKind.local, path: path.pathString))
+                }
+            }
+            
+            // Convert custom schemes. These are the explicitly specified schemes not the default
+            // ones generated automatically for all the targets.
+            var schemes = [SchemeOutputJSON]()
+            for scheme in project.schemes {
+                var testTargets = [String]()
+                if let testAction = scheme.testAction {
+                    for testTarget in testAction.targets {
+                        testTargets.append(testTarget.target.name)
+                    }
+                }
+                schemes.append(SchemeOutputJSON(name: scheme.name, testActionTargets: testTargets))
+            }
+            
+            // Convert targets.
+            let didDisableAutogeneratedSchemes = config.disableAutogeneratedSchemes
+            var targets = [TargetOutputJSON]()
+            for target in project.targets {
+                targets.append(TargetOutputJSON(name: target.name, product: target.product.rawValue))
+                
+                // For each target, a default scheme is generated. Include these in the schemes output.
+                // If the target's product is a testing one, include the target as test action.
+                if !didDisableAutogeneratedSchemes {
+                    switch target.product {
+                    case .uiTests, .unitTests:
+                        schemes.append(SchemeOutputJSON(name: target.name, testActionTargets: [target.name]))
+                    default:
+                        schemes.append(SchemeOutputJSON(name: target.name))
+                    }
+                }
+            }
+            
+            var projectJSON = ProjectOutputJSON(name: project.name, path: project.path.pathString)
+            if !packages.isEmpty {
+                projectJSON.packages = packages
+            }
+            if !schemes.isEmpty {
+                projectJSON.schemes = schemes
+            }
+            if !targets.isEmpty {
+                projectJSON.targets = targets
+            }
+            projects[project.path.pathString] = projectJSON
+        }
+        
+        return GraphOutputJSON(name: graph.name, path: graph.path.pathString, projects: projects)
+    }
+}
+
+public struct ProjectOutputJSON: Codable, Equatable {
```

Comment:
I'd create a new file for this one.

While you do that, I'd also create a new folder in `TuistGraph` such as `TuistGraph/Output`. Once we decide to extract the models into their own product, we can do that very easily by just moving all the files in that directory

Reply:
Sounds good!